luigi/task_register.py

```diff
@@ -62,12 +62,14 @@ def __new__(metacls, classname, bases, classdict):
 
         Also register all subclasses.
 
-        Set the task namespace to whatever the currently declared namespace is.
+        When the set or inherited namespace evaluates to ``None``, set the task namespace to
+        whatever the currently declared namespace is.
         """
-        if "task_namespace" not in classdict:
-            classdict["task_namespace"] = metacls._default_namespace
-
         cls = super(Register, metacls).__new__(metacls, classname, bases, classdict)
+
+        if getattr(cls, "task_namespace", None) is None:
```

Comment:
instead of a `getattr` call here, can you just set `task_namespace = None` on the base class? thanks